src/hotspot/share/code/nmethod.hpp

```diff
@@ -221,38 +223,24 @@ class nmethod : public CodeBlob {
   // Offset of the unwind handler if it exists
   int _unwind_handler_offset;
 
-  int _consts_offset;
-  int _stub_offset;
-  int _oops_offset;                       // offset to where embedded oop table begins (inside data)
-  int _metadata_offset;                   // embedded meta data table
-  int _scopes_data_offset;
-  int _scopes_pcs_offset;
-  int _dependencies_offset;
-  int _handler_table_offset;
-  int _nul_chk_table_offset;
+  uint16_t _skipped_instructions_size;
+
+  // _oops_offset == _data_offset,  offset where embedded oop table begins (inside data)
+  uint16_t _metadata_offset; // embedded meta data table
+  uint16_t _dependencies_offset;
+  uint16_t _scopes_pcs_offset;
+  int      _scopes_data_offset;
+  int      _handler_table_offset;
+  int      _nul_chk_table_offset;
```

Comment:
Changed data sections offsets base from `header_begin()` to `data_begin()`.
Note, `ScopesPcs` and `ScopesData` data could be > 64Kb.